### DISCUSSION TITLE
Update Maps package to use Azure Maps

### DISCUSSION THREAD

**Discussion by jfversluis** [CommunityToolkit/Maui#2123] [New Feature Discussions]
Bing Maps has stopped giving out API keys to new accounts and existing accounts will have until somewhere in 2025 until those keys are deprecated.

The replacement is Azure Maps. So the question arises: do we want to update to Azure Maps or just deprecate our Maps package?

The "native" WinUI [Map control](https://learn.microsoft.com/windows/apps/windows-app-sdk/stable-channel#winui-maps-control) is now also implemented and actually uses Azure Maps.

So we could either adopt that or see if we can do that directly in .NET MAUI since there is now a Map control on the underlaying platform, which there wasn't before when we implemented ours.

**Comment by VladislavAntonyuk** [comment]
If the control is not ready by .NET MAUI team in .NET 9, we can update our control with the new "native" WinUI Map Control

**Reply by jfversluis** [reply]
Unless we're going to implement it, I don't think it will be ready for .NET 9

**Reply by ljjandre** [reply]
@jfversluis will the .Net MAUI team be planning to implement a MAUI windows map control based on the new-ish WinUI map control? This is a major blocker for our app right now. Our UWP app uses heavy mapping functionality on windows for our customers using ruggedized windows laptops in the field to read meters while driving. We do not have a replacement solution for migrating our old Xamarin UWP app to the MAUI WinUI app because of this lack of a native map control on windows issue.

**Reply by mikelor** [reply]
This would be a big win. Ideally it would be part of the official Maui library.

I would also like to see about adding g new features to the Maui Map control like support for Custom Pins and Clustering. I realize that there are other libraries out there, but maybe we can look at enhancing the current Map control.

**Reply by jfversluis** [reply]
@ljjandre I don't think so. I had [another look in Oct](https://github.com/dotnet/maui/issues/12694#issuecomment-2449768174). There is indeed now a WinUI Map control, however it lacks basically [all basic functionality](https://github.com/microsoft/microsoft-ui-xaml/issues/3024#issuecomment-1932950478):

* Can't select map style (satellite, street, hybrid)
* No polylines/polygons
* Not able to show the users location
* Missing events for region changed, etc.
* Can't show traffic, etc.

So even if we were to implement it at this time it will be lacking a lot of features. To make this happen, the best thing you can do right now is upvote [issues](https://github.com/microsoft/microsoft-ui-xaml/issues?q=is%3Aissue%20state%3Aopen%20MapControl) for features that you need in the Windows MapControl or create new issues for the ones that aren't there yet.

Alternatives for now are:
* Mapsui
* Arcgis
* Syncfusion maps

**Reply by VladislavAntonyuk** [reply]
You can take a look how I implemented custom map in WorldExplorer app. It uses free OpenStreetMap and Leaflet, and looks the same on all platforms.

https://github.com/VladislavAntonyuk/WorldExplorer/tree/main/src/Client/Client/Controls/WorldExplorerMap
